core/src/main/java/org/kin/sendkin/core/store/RecipientContactsLocalStore.java

```diff
@@ -0,0 +1,35 @@
+package org.kin.sendkin.core.store;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+
+import org.kin.sendkin.core.model.RecipientContact;
+import org.kin.sendkin.core.view.Utils;
+
+import java.util.ArrayList;
+
+public class RecipientContactsLocalStore extends LocalStore implements RecipientContactsStore {
+
+    private final String KEY_ALL_CONTACTS = "KEY_ALL_CONTACTS";
+
+
+    public RecipientContactsLocalStore(@NonNull Context context, @NonNull String name) {
+        super(context, name);
+    }
+
+    @Override
+    public void save(ArrayList<RecipientContact> recipientContacts) {
+        putString(KEY_ALL_CONTACTS, Utils.parseToString(recipientContacts));
+
+    }
+
+    @NonNull
+    @Override
+    public ArrayList<RecipientContact> getRecipientContacts() {
+        String contactsStr = getString(KEY_ALL_CONTACTS);
+        if (contactsStr == null) {
+            return new ArrayList<>();
+        }
+        return Utils.parseContactsFromString(getString(KEY_ALL_CONTACTS));
```

Comment:
please handle errors in case of failed parsing, catch exceptions and propagate to UI level, crucial to prevent crashes in host apps in case something goes wrong (and it will, gson might do problems, proguard, etc.) can be forked to a different task. (phase 2)

Reply:
good point